### ISSUE TITLE
bug(cuda): wave-based >4096 batched SDPA diverges bitwise from per-position path at index 0 (AttentionBatchedWave, Coder-30B, laptop)

### ISSUE THREAD

**Issue by pekkah** [pekkah/SharpInference#419]
Found while running the full `Tests.ForwardPass` suite during #417 (laptop: Core Ultra 9 185H / RTX 4070 Laptop 8 GB, driver as of 2026-07-03):

```
CudaHybridBatchedPrefillTests.BatchedPrefill_Over4096_BitwiseMatchesSequential_Coder [FAIL]
  Wave-based >4096 batched SDPA diverges from the per-position path at index 0 (N=4231).
  AttentionBatchedWave host wiring must be bit-identical to the per-position loop.
```

**Verified pre-existing on unmodified master (9e877b9)** — reproduced identically via a clean `git stash` A/B, so it is not caused by the #417/#415 line of work. Model: `Qwen3-Coder-30B-A3B-Instruct-Q4_K_M.gguf` (no Q8_0 tensors).

Repro:
```
dotnet test tests/SharpInference.Tests.ForwardPass -c Release --filter "FullyQualifiedName~BatchedPrefill_Over4096_BitwiseMatchesSequential_Coder"
```
(~10.5 min on the laptop; needs the Coder GGUF on disk and CUDA.)

Observations / triage starters:

- Divergence at **index 0** of the final-token logits is systematic, not thermal or flaky-reduction noise.
- The test forces the multi-wave loop with `SHARPI_ATTN_WAVE_BUDGET_MB=8` on a >4096-token prompt — the wave path (`AttentionBatchedWave`, issue #118) only engages past the 4096-position shared-scores window, so shorter-prompt parity tests (which pass) never exercise it. The sibling `BatchedPrefill_BitwiseMatchesSequential_Coder` and the multi-chunk variant pass on the same box, which brackets the regression to the wave path specifically (host wiring or kernel), not the batched trunk generally.
- Unknown whether it reproduces on the desktop (RTX 4070 Ti) — flagged in the #416 validation protocol (see the addendum comment there). If it is laptop-only, suspicion shifts toward VRAM-pressure-dependent behavior (8 GB card, large ctx) in the wave scratch sizing/eviction rather than a plain indexing bug.
- Until fixed, this test + `VulkanInitTests.ComputeShaderVectorAdd` (no glslc installed) are the known-baseline failures on the laptop.

Related: #118 (AttentionBatchedWave), #416 (desktop validation trip that should try to reproduce it).
